// daypilot-pro-angular/esm2020/lib/daypilot.module.mjs
import { NgModule } from '@angular/core';
import { DayPilotCalendarComponent } from './daypilot-calendar.component';
import { DayPilotMonthComponent } from "./daypilot-month.component";
import { DayPilotNavigatorComponent } from "./daypilot-navigator.component";
import { DayPilotSchedulerComponent } from "./daypilot-scheduler.component";
import { DayPilotGanttComponent } from "./daypilot-gantt.component";
import { DayPilotKanbanComponent } from "./daypilot-kanban.component";
import { DayPilotQueueComponent } from "./daypilot-queue.component";
import { DayPilotModalComponent } from "./daypilot-modal.component";
import * as i0 from "@angular/core";
const COMPONENTS = [
    DayPilotCalendarComponent,
    DayPilotMonthComponent,
    DayPilotNavigatorComponent,
    DayPilotSchedulerComponent,
    DayPilotGanttComponent,
    DayPilotKanbanComponent,
    DayPilotQueueComponent,
    DayPilotModalComponent,
];
export class DayPilotModule {
}
DayPilotModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: DayPilotModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
DayPilotModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: DayPilotModule, declarations: [DayPilotCalendarComponent,
        DayPilotMonthComponent,
        DayPilotNavigatorComponent,
        DayPilotSchedulerComponent,
        DayPilotGanttComponent,
        DayPilotKanbanComponent,
        DayPilotQueueComponent,
        DayPilotModalComponent], exports: [DayPilotCalendarComponent,
        DayPilotMonthComponent,
        DayPilotNavigatorComponent,
        DayPilotSchedulerComponent,
        DayPilotGanttComponent,
        DayPilotKanbanComponent,
        DayPilotQueueComponent,
        DayPilotModalComponent] });
DayPilotModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: DayPilotModule, imports: [[]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: DayPilotModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        ...COMPONENTS
                    ],
                    imports: [],
                    exports: [
                        ...COMPONENTS
                    ]
                }]
        }] });
